src/core/qgsmaplayermodel.cpp

```diff
@@ -154,6 +154,11 @@ void QgsMapLayerModel::removeLayers( const QStringList &layerIds )
 
 void QgsMapLayerModel::addLayers( const QList<QgsMapLayer *> &layers )
 {
+  if ( layers.isEmpty() )
```

Comment:
@t-hey 

what's this change intended to address? I'm guessing it's unrelated to the mingw build, but was it a result of a bug you hit?

Reply:
Yes, Q_ASSERT triggers in the beginInsertRows statement in this procedure at startup.

At startup the input parameters for beginInsertRows, "first" is 0 and "last" is -1, then Q_ASSERT(last >= first) triggers in the [QAbstractItemModel](https://github.com/qt/qtbase/blob/5.9.3/src/corelib/itemmodels/qabstractitemmodel.cpp).

Also if mAllowEmpty was true at startup, it would by fine, and also the addition of "offset" should be only set on the "last" paramater of beginInsertRows statement. 

Then there is no need for checking isEmpty() on the incoming layers list.